### PR TITLE
style(sbb-button): button disabled state

### LINKED ISSUE #2636: sbb-button: adapt new disabled design

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #2636 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../docs/REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- new disabled state

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/core/styles/mixins/buttons.scss [hunk 1]
```diff
@@ -145,6 +147,9 @@
 
 // Disabled state for icon-button in form-field
 @mixin icon-button-disabled($button-selector) {
+  --sbb-button-border-width: #{functions.px-to-rem-build(1)};
```
Comment (jeripeierSBB) [inline src/components/core/styles/mixins/buttons.scss:150]:
```suggestion
  --sbb-button-border-width: var(--sbb-border-width-1x);
```

File: src/components/button/common/button-common.scss [hunk 2]
```diff
@@ -81,6 +82,8 @@ $icon-only: ':where([data-slot-names~=icon], [icon-name]):not([data-slot-names~=
 }
 
 :host([disabled]) {
+  --sbb-button-border-style: dashed;
```
Comment (jeripeierSBB) [inline src/components/button/common/button-common.scss:85]: Could you please align the variables to the "system" used in button-common: there should be a variable for every state. Moreover, the border color should be smoke (so the current outcome doesn't look correct).

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

**Comment by mcilurzo** [comment]
ux-review: the border color seems wrong. it sould be "smoke" in the default and negative variant.

File: src/components/button/common/button-common.scss [hunk 3]
```diff
@@ -54,7 +56,7 @@ $icon-only: ':where([data-slot-names~=icon], [icon-name]):not([data-slot-names~=
 
 :host([negative]) {
   --sbb-button-color-disabled-background: var(--sbb-color-anthracite);
-  --sbb-button-color-disabled-border: var(--sbb-color-granite);
+  --sbb-button-color-disabled-border: var(--sbb-color-smoke);
```
Comment (TomMenga) [inline src/components/button/common/button-common.scss:59]:
```suggestion
```
I think it's not needed anymore

**Review by TomMenga — APPROVED** [review]
Just a small thing.
The rest is lgtm 👍

**Comment by mcilurzo** [comment]
ux review: the border of the disabled state should be 1px

**Review by mcilurzo — APPROVED** [review]

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**